Helper/TbkResponseHelper.php

```diff
@@ -0,0 +1,197 @@
+                nl2br('• Monto:$ ' . $transactionResult->amount . ' ') .
+                nl2br('• Order de Compra: ' . $transactionResult->buyOrder . ' ') .
+                nl2br('• Fecha de la Transacción: ' . substr(self::utcToLocalDate($transactionResult->transactionDate), 0,10) . ' ') .
+                nl2br('• Hora de la Transacción: ' . substr(self::utcToLocalDate($transactionResult->transactionDate),11, 8). ' ') .
+                nl2br('• Tarjeta: **** **** **** ' . $transactionResult->cardNumber . '');
+        }
+
+        if (isset($transactionResult)) {
+            $message = "<h2>Transacci&oacute;n rechazada con {$product}</h2>
+                <div>
+                    • Respuesta de la Transacci&oacute;n: <b>{$transactionResult->responseCode}</b>
+                </div>
+                <div>
+                    • Monto:<b>$ {$transactionResult->amount}</b>
+                </div>
+                <div>
+                    • Order de Compra:<b> {$transactionResult->buyOrder}</b>
+                </div>
+                <div>
+                    • Fecha de la Transacci&oacute;n: <b>" . substr(self::utcToLocalDate($transactionResult->transactionDate), 0,10)."</b>
+                </div>
+                <div>
+                    • Hora de la Transacci&oacute;n: <b>" . substr(self::utcToLocalDate($transactionResult->transactionDate),11, 8)."</b>
+                </div>
+                <div>
+                    • Tarjeta: <b>**** **** **** {$transactionResult->cardNumber}</b>
+                </div>
+           ";
+
+        } else {
+            if ($transactionResult->status == 'ERROR') {
+                $error = $transactionResult->status;
+                $detail = isset($transactionResult->details[0]) ? $transactionResult->details[0] : 'Sin detalles';
+                $message = "<h2>Transacci&oacute;n fallida con {$product}</h2>
+                <div>
+                    <b>• Respuesta de la Transacci&oacute;n: </b>{$error}
+                </div>
+                <div>
+                    <b>• Mensaje: </b>{$detail}
+                </div>";
+            } else {
+                $message = '<h2>Transacci&oacute;n Fallida</h2>';
+            }
+        }
+
+        return $message;
+    }
+
+
+    public static function getOneclickDetails($transactionResult){
+        $details = $transactionResult->details;
+        foreach ($details as $detail) {
+            $transactionResult->amount = $detail->amount;
+            $transactionResult->authorizationCode = $detail->authorizationCode;
+            $transactionResult->buyOrder = $detail->buyOrder;
+            $transactionResult->installmentsNumber = $detail->installmentsNumber;
+            $transactionResult->paymentTypeCode = $detail->paymentTypeCode;
+            $transactionResult->responseCode = $detail->responseCode;
+            $transactionResult->status = $detail->status;
+        }
```

Comment:
this foreach isn't necessary, you can access directly to details on index [0].

Reply:
This foreach allows you to format transactionResult and use the same function for oneclick and webpay messages, avoiding code duplication